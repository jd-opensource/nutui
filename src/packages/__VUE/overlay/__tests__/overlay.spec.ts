import { mount } from '@vue/test-utils';
import OverLay from '../index.vue';

test('should change z-index when using z-index prop', () => {
  const wrapper = mount(OverLay, {
    props: {
      modelValue: true,
      zIndex: 99
    }
  });

  const overlay: any = wrapper.find('.nut-overlay');
  expect(overlay.element.style.zIndex).toEqual('99');
});

test('should change animation duration when using duration prop', () => {
  const wrapper = mount(OverLay, {
    props: {
      modelValue: true,
      duration: 1
    }
  });

  const overlay: any = wrapper.find('.nut-overlay');
  expect(overlay.element.style.animationDuration).toEqual('1s');
});

test('prop overlay-class test', () => {
  const wrapper = mount(OverLay, {
    props: {
      modelValue: true,
      overlayClass: 'testclass'
    }
  });

  const overlay: any = wrapper.find('.nut-overlay');
  expect(overlay.classes()).toContain('testclass');
});

test('prop close-on-click-overlay test', async () => {
  const wrapper = mount(OverLay, {
    props: {
      modelValue: true,
      closeOnClickOverlay: false
    }
  });
  await wrapper.trigger('click');
  const overlay: any = wrapper.find('.nut-overlay');
  expect(overlay.element.style.display).toEqual('none');
});

test('event click test', async () => {
  const wrapper = mount(OverLay, {
    props: {
      modelValue: true,
      closeOnClickOverlay: true
    }
  });

  const overlay: any = wrapper.find('.nut-overlay');
  await overlay.trigger('click');
  expect(wrapper.emitted('click')).toBeTruthy();
});

test('slots test', () => {
  const wrapper = mount(OverLay, {
    props: {
      modelValue: true
    },
    slots: {
      default: '这里是正文'
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});
