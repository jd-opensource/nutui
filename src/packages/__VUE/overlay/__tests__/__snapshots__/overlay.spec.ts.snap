// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`slots test 1`] = `
"<transition-stub name="overlay-fade" appear="false" persisted="true" css="true">
  <view class="nut-overlay" style="transition-duration: 0.3s; z-index: 2000;">这里是正文</view>
</transition-stub>"
`;
